ethcore/src/state_db.rs

```diff
@@ -39,22 +43,94 @@ pub struct StateDB {
 	account_cache: Arc<Mutex<AccountCache>>,
 	cache_overlay: Vec<(Address, Option<Account>)>,
 	is_canon: bool,
+	account_bloom: Arc<Mutex<Bloom>>,
 }
 
+pub const ACCOUNT_BLOOM_SPACE: usize = 1048576;
+pub const DEFAULT_ACCOUNT_PRESET: usize = 1000000;
+
+pub const ACCOUNT_BLOOM_HASHCOUNT_KEY: &'static[u8] = b"account_hash_count";
+
 impl StateDB {
+
+	pub fn check_bloom_exists(db: &Database) -> bool {
+		let hash_count_entry = db.get(DB_COL_ACCOUNT_BLOOM, ACCOUNT_BLOOM_HASHCOUNT_KEY)
+			.expect("Low-level database error");
+
+		hash_count_entry.is_some()
+	}
```

Comment:
this function is only used for the migration; maybe it should be implemented there? (especially as it relies on external constants)